packages/react-form-renderer/src/form-renderer/form-renderer.js

```diff
@@ -53,8 +53,8 @@ const FormRenderer = ({
     return !isFunc(onSubmit) ? undefined : onSubmit(values, {...formApi, fileInputs}, ...args);
   }, [onSubmit, fileInputs]);
 
-  const handleCancelCallback = useCallback((getState) => (...args) => {
-    return !isFunc(onCancel) ? void 0 : onCancel(getState().values, ...args);
+  const handleCancelCallback = useCallback((getState) => {
+    return !isFunc(onCancel) ? undefined : ((...args) => onCancel(getState().values, ...args));
```

Comment:
@Hyperkid123 Am I missing something? I thought I copied directly almost the exact thing that was there before. The useCallback initial return is called immediately before passing to the context provider.

https://github.com/data-driven-forms/react-forms/blob/3cfbaf6f88ba81a8aae94be2ffd9bcf2c59c1cc0/packages/react-form-renderer/src/form-renderer/form-renderer.js#L85

Reply:
The issue is that the `onCancel` passed to the renderer is always a function. (use call back will always return a function). Thus the cancel buttons show even if the user did not define any `onCancel` function. Even if you return `undefined` from the callback, it will generate a `() => undefined` function which is still going to trigger the conidtion.

Reply:
The implementation of the `handleCallback` was OK before.